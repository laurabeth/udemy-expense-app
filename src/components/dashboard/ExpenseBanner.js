import React from "react";

const ExpenseBanner = () => <div>Expense Banner</div>;

export default ExpenseBanner;
