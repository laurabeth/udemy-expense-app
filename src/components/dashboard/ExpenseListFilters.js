/* eslint-disable jsx-a11y/no-onchange */
import React from "react";
import { DateRangePicker } from "react-dates";
import { connect } from "react-redux";
import {
  setEndDate,
  setStartDate,
  setTextFilter,
  sortByAmountAscending,
  sortByDateAscending,
  sortByDateDescending,
  sortByAmountDescending,
} from "../../actions";

const { useState } = React;

export const ExpenseListFilters = ({
  filters,
  setEndDate,
  setStartDate,
  setTextFilter,
  sortByAmountAscending,
  sortByAmountDescending,
  sortByDateAscending,
  sortByDateDescending,
}) => {
  const [calendarFocused, setCalendarFocused] = useState(null);
  const handleChangeFilter = (e) => {
    setTextFilter(e.target.value);
  };

  const handleSelectSort = (e) => {
    switch (e.target.value) {
      case "date_asc":
        sortByDateAscending();
        break;
      case "date_desc":
        sortByDateDescending();
        break;
      case "amount_asc":
        sortByAmountAscending();
        break;
      case "amount_desc":
        sortByAmountDescending();
        break;
    }
  };

  const handleChangeFilterDates = ({ startDate, endDate }) => {
    setStartDate(startDate);
    setEndDate(endDate);
  };

  return (
    <div>
      <input onChange={handleChangeFilter} type="text" value={filters.text} />
      <select onChange={handleSelectSort} value={filters.sortBy}>
        <option value="amount_asc">Amount (Ascending)</option>
        <option value="amount_desc">Amount (Descending)</option>
        <option value="date_asc">Date (Ascending)</option>
        <option value="date_desc">Date (Descending)</option>
      </select>
      <DateRangePicker
        endDate={filters.endDate}
        endDateId="endDate"
        focusedInput={calendarFocused}
        isOutsideRange={() => false}
        numberOfMonths={1}
        onDatesChange={handleChangeFilterDates}
        onFocusChange={(focused) => setCalendarFocused(focused)}
        showClearDates
        startDate={filters.startDate}
        startDateId="startDate"
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

const mapDispatchToProps = (dispatch) => ({
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmountAscending: () => dispatch(sortByAmountAscending()),
  sortByAmountDescending: () => dispatch(sortByAmountDescending()),
  sortByDateAscending: () => dispatch(sortByDateAscending()),
  sortByDateDescending: () => dispatch(sortByDateDescending()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
