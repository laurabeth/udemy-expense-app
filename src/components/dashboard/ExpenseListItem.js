import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ amount, createdOn, description, id, note }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>Amount: {amount}</p>
      <p>Date: {createdOn.format("ddd, MMM, Do YYYY, LT")}</p>
      {note && <p>Note: {note}</p>}
    </div>
  );
};

export default ExpenseListItem;
