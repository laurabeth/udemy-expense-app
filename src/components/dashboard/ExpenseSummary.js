import numeral from "numeral";
import React from "react";
import { connect } from "react-redux";
import getExpensesTotal from "../../selectors/expenses-total";
import { getVisibleExpenses } from "../../selectors";

export const ExpenseSummary = ({ numberOf, total }) => (
  <div>{`Viewing ${numberOf} expenses, totaling ${numeral(total / 100).format(
    "$0,0.00",
  )}`}</div>
);

const mapStateToProps = (state) => {
  const expenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    numberOf: expenses.length,
    total: getExpensesTotal(expenses),
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
