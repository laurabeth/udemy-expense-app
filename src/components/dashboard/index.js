export { default as ExpenseDashboard } from "./ExpenseDashboard";
export { default as ExpenseList } from "./ExpenseList";
export { default as ExpenseListFilters } from "./ExpenseListFilters";
export { default as ExpenseListItem } from "./ExpenseListItem";
