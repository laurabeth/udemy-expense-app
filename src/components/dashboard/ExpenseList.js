import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import { getVisibleExpenses } from "../../selectors";

export const ExpenseList = (props) => {
  return (
    <>
      {props.expenses.length === 0 ? (
        <p>No expenses found.</p>
      ) : (
        <div>
          {props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })}
        </div>
      )}
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseList);
