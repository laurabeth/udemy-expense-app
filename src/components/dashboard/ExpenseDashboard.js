import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashboard = () => {
  return (
    <div>
      <ExpenseList />
      <ExpenseListFilters />
    </div>
  );
};

export default ExpenseDashboard;
