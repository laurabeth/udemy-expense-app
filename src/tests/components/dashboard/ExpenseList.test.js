import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../../components/dashboard/ExpenseList";
import expensesMock from "../../fixtures/expenses";

/* eslint-disable no-undef */
describe("expense list", () => {
  it("renders with data", () => {
    const wrapper = shallow(<ExpenseList expenses={expensesMock} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders without data", () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
