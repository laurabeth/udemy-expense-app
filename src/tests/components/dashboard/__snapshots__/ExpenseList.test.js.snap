// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expense list renders with data 1`] = `
<Fragment>
  <div>
    <ExpenseListItem
      amount={50}
      createdOn={"1970-01-01T00:00:00.000Z"}
      description="gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={485}
      createdOn={"1969-12-31T00:00:00.000Z"}
      description="green rock candy"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={800}
      createdOn={"1970-01-02T00:00:00.000Z"}
      description="chocolate"
      id="3"
      key="3"
      note=""
    />
  </div>
</Fragment>
`;

exports[`expense list renders without data 1`] = `
<Fragment>
  <p>
    No expenses found.
  </p>
</Fragment>
`;
